### PR TITLE
VS-24: Initial stable version

### PR DESCRIPTION
**Pull request by BorisDog**

### CODE REVIEW

File: tests/MongoDB.Analyzer.Tests/Infrastructure/TestCasesProvider.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                Where(d => d.Location.IsInSource && d.Severity < DiagnosticSeverity.Error).
+                Select(d =>
+                    (Diagnostic: d,
+                     MethodNode: d.Location.SourceTree.
+                        GetRoot().
+                        FindNode(d.Location.SourceSpan).
+                        Ancestors().
+                        OfType<MethodDeclarationSyntax>().
+                        First())).
+                ToArray();
```
Comment (JamesKovacs) [inline tests/MongoDB.Analyzer.Tests/Infrastructure/TestCasesProvider.cs:38]: Our multi-line coding style has dot operator at the beginning of the line.

File: tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqExpressionsBasicTestCases.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+﻿using MongoDB.Driver;
+using MongoDB.Driver.Linq;
+using System.Linq;
+
+namespace MongoDB.Analyzer.Tests.Common.TestCases.Linq
+{
+    public sealed class LinqExpressionsBasicTestCases : TestCasesBase
+    {
+        // MongoLinq2MQ
+        // aggregate([{ "$match" : { "Name" : "Bob", "Age" : { "$gt" : 16, "$lte" : 21 } } }])
```
Comment (JamesKovacs) [inline tests/MongoDB.Analyzer.Tests.Common.TestCases/Linq/LinqExpressionsBasicTestCases.cs:10]: As discussed in Slack, these comments would be better expressed as custom attributes. I believe you already changed this in a later PR, but wanted to confirm.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Great first cut. Incorporating this feedback isn't a blocker to the beta release. It can be done post-beta1.

File: src/MongoDB.Analyzer/Core/ReferencesProvider.cs [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+                else
+                {
+                    return metadataReferencesSource.
+                        OfType<PortableExecutableReference>().
+                        Where(r => r.FilePath.Contains(NetstandardDll)).
+                        FirstOrDefault().FilePath;
+                }
+            }
+
+            string GetMongoDBDriverNetFrameworFullPath(string assemblyName)
```
[the remark this answers is not in the document]
Comment (BorisDog) [reply src/MongoDB.Analyzer/Core/ReferencesProvider.cs:102]: Gone

File: src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.CodeAnalysis;
+using System.Linq;
+
+namespace MongoDB.Analyzer.Core
+{
+    internal static class SymbolExtensions
+    {
+        public static bool ImplementsInterface(this ITypeSymbol typeSymbol, string interfaceName) =>
```
[the remark this answers is not in the document]
Comment (BorisDog) [reply src/MongoDB.Analyzer/Core/Utilities/SymbolExtensions.cs:8]: Good points:
1. Done
2. Done
3. We only need a type that implements this interface, or is the actual interface. In the case of a base class, all interfaces will be listed in `typeSymbol.Interfaces`. Changed the name to `ImplementsOrIsInterface`

**Review by JamesKovacs — APPROVED** [review]
LGTM.

N.B. The requested changes were made in another branch, which is why they're not showing up in this PR.

**Comment by BorisDog** [comment]
Merge in [other PR](https://github.com/mongodb/mongo-csharp-analyzer/pull/3)

**Pull request closed without merging**